src/settings.js

```diff
@@ -404,7 +404,6 @@ var EMSCRIPTEN_VERSION = ''; // this will contain the emscripten version. you sh
                              // RETAIN_COMPILER_SETTINGS
 var OPT_LEVEL = 0;           // this will contain the optimization level (-Ox). you should not modify it.
 var DEBUG_LEVEL = 0;         // this will contain the debug level (-gx). you should not modify it.
-var PROFILING_FUNCS = 0;     // Whether we are profiling functions. you should not modify it.
```

Comment:
I like this change; PROFILING_FUNCS was confusing before, as it seemed to be used to preserve the names even in cases that had nothing to do with profiling.

Reply:
The reason why `--profiling-funcs` was added was that earlier one would have to use `-g` in optimized builds to do a build that was as optimized as possible, with the single exception that one would want to retain symbol information so that the resulting build was readily profilable in browser profilers. But in Emscripten `-g` (at the time it was added, still true today?) implies other things that prevented some optimizations from taking place, and/or added extra code size elsewhere, so having an option that only added function symbols for profiling, but "optimize and minimize fully otherwise" made sense.

I'm not sure if this PR will allow us to remove this option now? Do we have an equivalent of a `-O3 --profiling-funcs` build now? That is my usual go-to mode that I pretty much always do optimized builds with when working on performance. It gets the fastest and smallest code that I can drive through Gecko Profiler and other in-browser profilers. Zen Garden asm.js & wasm builds at http://mzl.la/webassemblydemo are shipped with that combination, so that browser teams developing wasm support can do profiles on it. My understanding is that doing a `-O3 -g` (or `-O3 -g1`) is still suboptimal compared to `-O3 --profiling-funcs` since it adds more debug features than are needed for this use case?

Reply:
Err, searching through history, I see I'm confusing `--profiling-funcs` with `-s PROFILING_FUNCS=1`, the latter which was added for wasm backend specifically in https://github.com/kripken/emscripten/commit/663fd4213575c8d52d799c0b1a9d95e182f6687f. I'm not sure about the utilization of `-s PROFILING_FUNCS=1`, could `--profiling-funcs` just be reused for wasm backend as well?

Sidenote: one might ponder that perhaps we should have an "optimize for profiling" mode such as `-Op` for this. But turns out something like that is not as nice as `--profiling-funcs`, because having a separate flag for just adding the needed bits to be able to do profiling allows pairing with different optimization levels, e.g. a `--profiling-funcs -O1` or `--profiling-funcs -O2` to compare how different opts levels affect performance. `-O' flags are usually mutually exclusive, so `-Op -O1` vs `-Op -O2` would probably be expected to be overriding and not combining.